truelearn/models/_learner_model.py

```diff
@@ -66,28 +66,25 @@ class LearnerMetaModel:
         >>> # construct an empty learner meta model
         >>> LearnerMetaModel()  # doctest:+ELLIPSIS
         LearnerMetaModel(learner_novelty=..., learner_interest=..., \
-novelty_weight={'mean': 0.0, 'variance': 1.0}, interest_weight=..., bias_weight=...)
+novelty_weight=Weights(mean=0, variance=0.5), interest_weight=..., bias_weight=...)
         >>> # construct a learner meta model with given learner models
         >>> learner_novelty = LearnerModel()
         >>> learner_interest = LearnerModel()
         >>> LearnerMetaModel(learner_novelty=learner_novelty,\
 learner_interest=learner_novelty)
         LearnerMetaModel(learner_novelty=..., learner_interest=..., \
-novelty_weight={'mean': 0.0, 'variance': 1.0}, interest_weight=..., bias_weight=...)
+novelty_weight=Weights(mean=0, variance=0.5), interest_weight=..., bias_weight=...)
         >>> # construct a learner meta model with custom weights
-        >>> LearnerMetaModel(bias_weight={"mean": 1.0, "variance": 2.0})
+        >>> bias_weight = LearnerMetaModel.Weights(mean=1.0, variance=2.0)
+        >>> LearnerMetaModel(bias_weight=bias_weight)
         LearnerMetaModel(learner_novelty=..., learner_interest=..., \
-novelty_weight=..., interest_weight=..., bias_weight={'mean': 1.0, 'variance': 2.0})
+novelty_weight=..., interest_weight=..., bias_weight=Weights(mean=1.0, variance=2.0))
     """
 
+    Weights = collections.namedtuple("Weights", "mean variance", defaults=[0, 0.5])
+
     learner_novelty: LearnerModel = dataclasses.field(default_factory=LearnerModel)
     learner_interest: LearnerModel = dataclasses.field(default_factory=LearnerModel)
-    novelty_weight: Dict[str, float] = dataclasses.field(
-        default_factory=lambda: {"mean": 0.0, "variance": 1.0}
-    )
-    interest_weight: Dict[str, float] = dataclasses.field(
-        default_factory=lambda: {"mean": 0.0, "variance": 1.0}
-    )
-    bias_weight: Dict[str, float] = dataclasses.field(
-        default_factory=lambda: {"mean": 0.0, "variance": 1.0}
-    )
+    novelty_weight: Weights = Weights()
```

Comment:
Loads of parameter mean unfilled and variance unfilled errors for 88 to 90

Reply:
This is weird. I don't have this warning. We could use `Weights' in this way, since we have defined `defaults'.

Reply:
Yea turns out to be something Pycharm doesnt want to fix:
https://youtrack.jetbrains.com/issue/PY-40405

There is a workaround, but im wondering if its even worth it

Reply:
As we are using `typing`, let's fix it.